packages/swapper/src/swappers/thorchain/utils/getLimit/getLimit.ts

```diff
@@ -0,0 +1,47 @@
+import { Asset } from '@shapeshiftoss/asset-service'
+
+import { SwapError, SwapErrorTypes } from '../../../../api'
+import { bn, bnOrZero, fromBaseUnit, toBaseUnit } from '../../../utils/bignumber'
+import { ThorchainSwapperDeps } from '../../types'
+import { THORCHAIN_FIXED_PRECISION } from '../constants'
+import { getTradeRate } from '../getTradeRate/getTradeRate'
+
+export const getLimit = async ({
+  sellAsset,
+  buyAsset,
+  sellAmount,
+  deps,
+  slippageTolerance,
+  tradeFee
+}: {
+  buyAssetId: string
+  destinationAddress: string
+  sellAsset: Asset
+  buyAsset: Asset
+  sellAmount: string
+  deps: ThorchainSwapperDeps
+  slippageTolerance: string
+  tradeFee: string
+}): Promise<string> => {
+  const tradeRate = await getTradeRate(sellAsset, buyAsset.assetId, sellAmount, deps)
+  const expectedBuyAmountPrecision8 = toBaseUnit(
+    fromBaseUnit(bnOrZero(sellAmount).times(tradeRate), sellAsset.precision),
+    THORCHAIN_FIXED_PRECISION
+  )
+
+  const isValidSlippageRange =
+    bnOrZero(slippageTolerance).gte(0) && bnOrZero(slippageTolerance).lte(1)
+  if (bnOrZero(expectedBuyAmountPrecision8).lt(0) || !isValidSlippageRange)
+    throw new SwapError('[getThorTxInfo]: bad expected buy amount or bad slippage tolerance', {
+      code: SwapErrorTypes.BUILD_TRADE_FAILED,
+      details: { expectedBuyAmountPrecision8, slippageTolerance }
+    })
+
+  const tradeFeePrecision8 = toBaseUnit(bnOrZero(tradeFee), THORCHAIN_FIXED_PRECISION)
+
+  return bnOrZero(expectedBuyAmountPrecision8)
+    .times(bn(1).minus(slippageTolerance))
```

Comment:
Since we're already in the bignumber.js world
```suggestion
    .times(1.minus(slippageTolerance))
```

Reply:
i dont think that will work